rllib/examples/inference/policy_inference_after_training_w_connector.py

```diff
@@ -0,0 +1,265 @@
+    AddStatesFromEpisodesToBatch,
+    BatchIndividualItems,
+    NumpyToTensor,
+)
+from ray.rllib.core import DEFAULT_MODULE_ID
+from ray.rllib.core.columns import Columns
+from ray.rllib.core.rl_module.rl_module import RLModule
+from ray.rllib.env.single_agent_episode import SingleAgentEpisode
+from ray.rllib.examples.envs.classes.stateless_cartpole import StatelessCartPole
+from ray.rllib.utils.framework import try_import_torch
+from ray.rllib.utils.metrics import (
+    ENV_RUNNER_RESULTS,
+    EPISODE_RETURN_MEAN,
+)
+from ray.rllib.utils.test_utils import (
+    add_rllib_example_script_args,
+    run_rllib_example_script_experiment,
+)
+from ray.tune.registry import get_trainable_cls, register_env
+
+torch, _ = try_import_torch()
+
+
+def _env_creator(cfg):
+    return StatelessCartPole(cfg)
+
+
+register_env("stateless-cart", _env_creator)
+
+
+parser = add_rllib_example_script_args(default_reward=200.0)
+parser.set_defaults(
+    # Make sure that - by default - we produce checkpoints during training.
+    checkpoint_freq=1,
+    checkpoint_at_end=True,
+    # Use StatelessCartPole by default.
+    env="stateless-cart",
+)
+parser.add_argument(
+    "--explore-during-inference",
+    action="store_true",
+    help="Whether the trained policy should use exploration during action "
+    "inference.",
+)
+parser.add_argument(
+    "--num-episodes-during-inference",
+    type=int,
+    default=10,
+    help="Number of episodes to do inference over (after restoring from a checkpoint).",
+)
+
+
+if __name__ == "__main__":
+    args = parser.parse_args()
+
+    assert (
+        args.enable_new_api_stack
+    ), "Must set --enable-new-api-stack when running this script!"
+
+    base_config = (
+        get_trainable_cls(args.algo)
+        .get_default_config()
+        .training(
+            num_sgd_iter=6,
+            lr=0.0003,
+            vf_loss_coeff=0.01,
+        )
+        # Add an LSTM setup to the default RLModule used.
+        .rl_module(model_config_dict={"use_lstm": True})
+    )
+
+    print("Training LSTM-policy until desired reward/timesteps/iterations. ...")
+    results = run_rllib_example_script_experiment(base_config, args)
+
+    print("Training completed. Creating an env-loop for inference ...")
+
+    print("Env ...")
+    env = _env_creator(base_config.env_config)
+
+    # We build the env-to-module pipeline here manually, but feel also free to build it
+    # through the even easier:
+    # `env_to_module = base_config.build_env_to_module_connector(env=env)`, which will
+    # automatically add all default pieces necessary (for example the
+    # `AddStatesFromEpisodesToBatch` component b/c we are using a stateful RLModule
+    # here).
+    print("Env-to-module ConnectorV2 ...")
+    env_to_module = EnvToModulePipeline(
+        input_observation_space=env.observation_space,
+        input_action_space=env.action_space,
+        connectors=[
+            AddObservationsFromEpisodesToBatch(),
```

Comment:
Here we need more documentation which connectors are needed. For the module we get its inputs from the input specs. 
imo this example is still an easy one b/c the connectors are stateless themselves. This could be done w/o connectors easily but with a mean_std filter we need the actual state of the connector.

Reply:
Agreed, this example wouldn't work with stateful connectors.
I'll add more documentation to the examples. Need to write all the ConnectorV2 docs still anyways :)